src/util/permissionIdpGroups.tsx

```diff
@@ -21,3 +31,72 @@ export const testDuplicateIdpGroupName = (
     },
   ];
 };
+
+type DeleteIdpGroupsProps = {
+  idpGroups: IdpGroup[];
+  onSuccess?: () => void;
+};
+
+export const useDeleteIdpGroups = ({
+  idpGroups,
+  onSuccess,
+}: DeleteIdpGroupsProps) => {
+  const queryClient = useQueryClient();
+  const notify = useNotify();
+  const toastNotify = useToastNotification();
+  const [submitting, setSubmitting] = useState(false);
+  const { canDeleteIdpGroup } = useIdpGroupEntitlements();
+
+  const restrictedGroups: IdpGroup[] = [];
+  const deletableGroups: IdpGroup[] = [];
+  idpGroups.forEach((group) => {
+    if (canDeleteIdpGroup(group)) {
+      deletableGroups.push(group);
+    } else {
+      restrictedGroups.push(group);
+    }
+  });
+
+  const hasOneGroup = deletableGroups.length === 1;
+
+  const handleDeleteIdpGroups = () => {
+    setSubmitting(true);
+
+    const successMessage = hasOneGroup ? (
+      <>
+        IDP group{" "}
+        <ResourceLabel bold type="idp-group" value={deletableGroups[0].name} />{" "}
+        deleted.
+      </>
+    ) : (
+      `${deletableGroups.length} IDP groups deleted.`
+    );
+
+    deleteIdpGroups(deletableGroups.map((group) => group.name))
+      .then(() => {
+        void queryClient.invalidateQueries({
+          queryKey: [queryKeys.idpGroups],
+        });
+        toastNotify.success(successMessage);
+        if (onSuccess) {
+          onSuccess();
+        }
+      })
+      .catch((e) => {
+        notify.failure(
+          `${pluralize("IDP group", deletableGroups.length)} deletion failed`,
+          e,
+        );
+      })
+      .finally(() => {
+        setSubmitting(false);
+      });
+  };
+
+  return {
+    handleDeleteIdpGroups,
+    submitting,
+    restrictedGroups,
+    deletableGroups,
```

Comment:
```suggestion
    deleteIdpGroups,
    isDeleting,
    restrictedIdpGroups,
    deletableIdpGroups,
```
Suggestions for correct naming.

Reply:
done :+1: